src/entities/task/ui.tsx

```diff
@@ -1,16 +1,66 @@
-import { useSelector } from "react-redux";
+import { SyntheticEvent, useEffect, useState } from "react";
+import { useDispatch, useSelector } from "react-redux";
 
-import { selectTasks, Task } from "./model";
+import { deleteTask, selectAll, Task, updateTask } from "./model";
 
-const TaskItem = ({ title }: Omit<Task, "completed" | "id">) => <div>{title}</div>;
+const TaskItem = ({ title, id }: Omit<Task, "completed">) => {
+    const dispatch = useDispatch();
+    const [text, setText] = useState("");
+    const [isEditing, setIsEditing] = useState(false);
+    const handleClick = (id: string) => {
+        dispatch(deleteTask(id));
+    };
+
+    const handleChange = (e: SyntheticEvent) => {
+        const target = e.target as HTMLInputElement;
+        setText(target.value);
+    };
+
+    const handleFocus = (e: SyntheticEvent) => {
+        setIsEditing(true);
+        const target = e.target as HTMLInputElement;
+        setText(target.value);
+    };
+
+    const handleBlur = (e: SyntheticEvent) => {
+        const target = e.target as HTMLInputElement;
+        setText(target.value);
+        setIsEditing(false);
+        dispatch(
+            updateTask({
+                id,
+                changes: { title: text },
+            }),
+        );
+    };
+
+    useEffect(() => {
+        setText(title);
+    }, [title]);
```

Comment:
It's redundant. You can provide the title as an initial value of state
` const [text, setText] = useState(title);`